pkgs/development/python-modules/scalene/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, setuptools
+, setuptools-scm
+, wheel
+, cloudpickle
+, cython
+, jinja2
+, psutil
+, pynvml
+, pythonOlder
+, rich
+}:
+
+buildPythonPackage rec {
+  pname = "scalene";
+  version = "1.5.35";
+  pyproject = true;
+  disabled = pythonOlder "3.9";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-3s0KN06RivCN4G5/LZf+r7GtAlKwqRuZwUtrY2T2Bew=";
+  };
+
+  nativeBuildInputs = [
+    cython
+    setuptools
+    setuptools-scm
+    wheel
+  ];
+
+  propagatedBuildInputs = [
+    cloudpickle
+    jinja2
+    psutil
+    pynvml
+    rich
+  ];
+
+  pythonImportsCheck = [ "scalene" ];
+
+  meta = with lib; {
+    description = "Scalene: A high-resolution, low-overhead CPU, GPU, and memory profiler for Python with AI-powered optimization suggestions";
+    homepage = "https://pypi.org/project/scalene/1.5.35/";
+    changelog = "https://github.com/plasma-umass/scalene/releases";
```

Comment:
When possible include the relevant ref e.g. "https://github.com/plasma-umass/scalene/releases/v${version}" (presumably to help automated tooling)